### ISSUE TITLE
Accessibility and Usability Issues With VoiceOver On Mac OS

### ISSUE THREAD

**Issue by Rogue-Git-Dev** [transmission/transmission#4207]
### What is the issue?

Ever since I used Transmission years ago it has been littered with a myriad of accessibility issues related to VoiceOver on Mac OS. I alternated amongst other torrent clients and have used Transmission for quite sometime now – especially now that version 4 is released for there is hope these issues will be fixed once and for all. All of the usability pitfalls do not come to mind now but here are the most prominent issues.

* Tables in Transmission have blank cells and cells with unlabelled images. The most apodeictic example is the table with the torrents. Each row has six columns presumably; from left to right VoiceOver reads: image, blank, image, blank, image blank. I must press ⌘i and navigate to the first item in the info window to know the torrent – which serves as a segue to the next issue. Before that issue, another very useful exemplar is the table which shows trackers and their tiers. This is quite useless for the only information VoiceOver says is the tier and its number followed by blank separating each tier. The table which shows files is the worst offender. VoiceOver fails to say anything – not even blank – but simply unlabelled tick boxes and their states [either ticked or not ticked]. It is litterally impossible to select certain files in torrents for the entire functionality is unusable for VoiceOver users.
* Unlabelled images, buttons, tick boxes, and text fields abound. Recently a commit was made for image descriptions. This likely was not for Mac OS for the same images have not got labels. Every images needs labels and perhaps optional help tags / tooltips but definitely useful markup. Moreover, text fields generally lack labels but rather labels somewhere before them which to me makes no sense. Many buttons are unlabelled. The info window is a good example. The buttons to change tabs[?] that display information are all unlabelled thus hassle increases just to know which tab displays what information. I am sure other windows contain unlabelled buttons. The main window contains two nonsensically labelled images before the [unlabelled] dynamic text that are the numbers for upload and download speeds. The order of these images was changed in a 4.9.0 nightly build some time after the public beta was released. I know not which arrangement of the images made sense for neither of the images serve purpose to me. Perhaps useless images can be hidden from VoiceOver altogether.

These are only the most prominent problems that come to mind at this moment which make Transmission so unpleasant and difficult to use. There are other cosmetic ‘bugs’ as well I am sure. I understand to make Transmission equally usable to me as all of its sighted users will not be an overnight endeavour but I am more than obliged to test any builds and any specific changes. The time has come for it has been years overdue; Transmission must be the most VoiceOver accessible and usable torrent client for Mac OS.

### Which application of Transmission?

macOS app

### Which version of Transmission?

every single build since the beginning of time

**Comment by Rogue-Git-Dev** [comment]
Oh right, I forgot I submitted that for that was quite sometime ago. This duplicate may be more verbose.

**Comment by mikedld** [comment]
Since you authored both, please close [any] one and leave another.

**Comment by Coeur** [comment]
I did took a look at it last week when I did #4149, but that was all I could do.
If someone is knowledgeable in supporting accessibility features, then help is welcome.

**Comment by Rogue-Git-Dev** [comment]
If anyone wishes to test Transmission with VoiceOver enabled, I have got some suggestions to make it much easier to navigate beginning with enabling Quick Nav using both left and right arrow keys. The left and right arrow keys are used primarily to navigate.

**Comment by Rogue-Git-Dev** [comment]
Build 9125 fixed the table issue completely. The only thing wrong now is the buttons for each torrent remain unlabelled. Finally I can navigate the table and interact with items within the table. I know labels for buttons can be added relatively easily. This is such a significant update for accessibility.

**Comment by GaryElshaw** [comment]
Is this something that can be added @sweetppro?
